python/cuml/preprocessing/encoders.py

```diff
@@ -0,0 +1,291 @@
+                           "to be dropped, but were not found in the encoder "
+                           "categories.".format(feature))
+                    raise ValueError(msg)
+                cats = Series(cats)
+                idx = cats.isin(self.drop[feature])
+                drop_idx[feature] = cp.asarray(cats[idx].index)
+            return drop_idx
+        else:
+            msg = ("Wrong input for parameter `drop`. Expected "
+                   "'first', None or a dict, got {}")
+            raise ValueError(msg.format(type(self.drop)))
+
+    def fit(self, X):
+        """
+        Fit OneHotEncoder to X.
+        Parameters
+        ----------
+        X : cuDF.DataFrame
+            The data to determine the categories of each feature.
+        Returns
+        -------
+        self
+        """
+        self._validate_keywords()
+        if type(self.categories) is str and self.categories == 'auto':
+            self._encoders = {
+                feature: LabelEncoder(handle_unknown=self.handle_unknown).fit(
+                    X[feature])
+                for feature in X.columns
+            }
+        else:
+            self._encoders = dict()
+            for feature in self.categories.columns:
+                le = LabelEncoder(handle_unknown=self.handle_unknown)
+                self._encoders[feature] = le.fit(self.categories[feature])
+                if self.handle_unknown == 'error':
+                    if not X[feature].isin(self.categories[feature]).all():
+                        msg = ("Found unknown categories in column {0}"
+                               " during fit".format(feature))
+                        raise KeyError(msg)
+
+        self.drop_idx_ = self._compute_drop_idx()
+        self._fitted = True
+        return self
+
+    def fit_transform(self, X):
+        """
+        Fit OneHotEncoder to X, then transform X.
+        Equivalent to fit(X).transform(X).
+
+        Parameters
+        ----------
+        X : cudf.DataFrame
+            The data to encode.
+        Returns
+        -------
+        X_out : sparse matrix if sparse=True else a 2-d array
+            Transformed input.
+        """
+        return self.fit(X).transform(X)
+
+    @with_cupy_rmm
+    def _one_hot_encoding(self, feature, X):
+        encoder = self._encoders[feature]
+
+        col_idx = encoder.transform(X).to_gpu_array(fillna="pandas")
+        col_idx = cp.asarray(col_idx)
+
+        ohe = cp.zeros((len(X), len(encoder.classes_)), dtype=self.dtype)
+        # Filter out rows with null values
+        idx_to_keep = col_idx > -1
+        row_idx = cp.arange(len(ohe))[idx_to_keep]
+        col_idx = col_idx[idx_to_keep]
+        ohe[row_idx, col_idx] = 1
+
+        if self.drop_idx_ is not None:
+            drop_idx = self.drop_idx_[feature]
+            mask = cp.ones((ohe.shape[1]), dtype=cp.bool)
+            mask[drop_idx] = False
+            ohe = ohe[:, mask]
```

Comment:
The above can be replaced with one line:
`ohe = ohe[:, self.drop_idx_[feature]]